compiler/rustc_codegen_llvm/src/asm.rs

```diff
@@ -381,13 +389,119 @@ impl<'tcx> AsmCodegenMethods<'tcx> for CodegenCx<'_, 'tcx> {
         operands: &[GlobalAsmOperandRef<'tcx>],
         options: InlineAsmOptions,
         _line_spans: &[Span],
+        instance: Instance<'tcx>,
     ) {
         let asm_arch = self.tcx.sess.asm_arch.unwrap();
 
         // Default to Intel syntax on x86
         let intel_syntax = matches!(asm_arch, InlineAsmArch::X86 | InlineAsmArch::X86_64)
             && !options.contains(InlineAsmOptions::ATT_SYNTAX);
 
+        // Convert all operands to string interpolations
```

Comment:
I think it would be good to take this opportunity and move this into cg_ssa and and add some extra trait methods for the parts that are different. Similar to what I did in https://github.com/rust-lang/rust/pull/142960

Reply:
GCC 15+ has ability for file-level assembly to reference symbols and constants, so I think codegen_llvm and codegen_gcc may eventually diverge and use different approaches.